Update architecture.md (#4642)
grafana/agent: docs/sources/operator/architecture.md

The shard number is not added as a label, as sharding is designed to be
transparent on the receiver end.

> ## Enabling sharding and replication

Reviewer: ```suggestion
## Enable sharding and replication
```

We should use the imperative (Enable) instead of the gerund (Enabling).

> To enable sharding and replication, you need to set the `shards` and `replicas` properties in the Grafana Agent configuration file. For example, the following configuration file would shard the data into 3 shards and replicate each shard to 2 other Grafana Agent instances:

Reviewer: ```suggestion
To enable sharding and replication, you must set the `shards` and `replicas` properties in the Grafana Agent configuration file. For example, the following configuration file would shard the data into three shards and replicate each shard to two other Grafana Agent instances:
```

Single digit numbers should be spelled out: https://developers.google.com/style/numbers

```
shards: 3
replicas: 2
```

You can also enable sharding and replication by setting the `shards` and `replicas` arguments when you start the Grafana Agent. 

### Examples

> Here are some examples of how to enable sharding and replication in a Kubernetes environment:

Reviewer: ```suggestion
The following examples show you how to enable sharding and replication in a Kubernetes environment.
```

> * To shard the data into 3 shards and replicate each shard to 2 other Grafana Agent instances, you would use the following deployment manifest:

Reviewer: ```suggestion
* To shard the data into three shards and replicate each shard to two other Grafana Agent instances, you would use the following deployment manifest:
```

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: grafana-agent
spec:
  replicas: 3
  selector:
    matchLabels:
      app: grafana-agent
  template:
    metadata:
      labels:
        app: grafana-agent
    spec:
      containers:
      - name: grafana-agent
        image: grafana/agent:latest
        args:
        - "--shards=3"
        - "--replicas=2"

> ```

Reviewer: For alignment, this (and the following example code block) should be indented to align with the first char of the bullet the precedes it. This will align the code block with the bullet text.

> * To shard the data into 10 shards and replicate each shard to 3 other Grafana Agent instances, you would use the following deployment manifest:

Reviewer: ```suggestion
* To shard the data into 10 shards and replicate each shard to three other Grafana Agent instances, you would use the following deployment manifest:
```
